### PR TITLE
✨ [source-chargebee] Allow low-code incremental streams to run within the Concurrent CDK framework

### PR DESCRIPTION
**Pull request by brianjlai**
## What

Adds support for an incremental stream's partitions to run in parallel with each other.

## How

Processing low-code incremental streams within the concurrent framework was introduced in https://github.com/airbytehq/airbyte/pull/46662 and version `6.0.0` of the CDK. 

This PR upgrades the version to the latest to take advantage of concurrent processing. It also makes the necessary updates to `source.py` and `run.py` to pass incoming config/catalog/state when instantiating a source. This is a requirement of the concurrent CDK.

## User Impact

Technically none as all the backend processing changes shouldn't affect customers and we use default concurrency levels. It does however put some customers at a higher chance of encountering a rate limit.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

We emit state in the legacy sequential format so that we can roll this back if needed. Once we start emitting concurrent state, it will be much harder to migrate backwards

### CODE REVIEW

**Comment by brianjlai** [comment]
Regression test report:

ran the following two scenarios which were similar to our testing of the CDK fix earlier this week:

- `customers` incremental stream without state to test date interval partitioning - https://github.com/airbytehq/airbyte/actions/runs/11608326347/job/32323364883
  - There is one extra record, which is something that we did notice in prior testing, but when using concurrency == 1 instead of 3, these mismatches tended to go away so this might be a timing issue since the order of partitions and speed is changign
- many streams of different sizes with incoming state - https://github.com/airbytehq/airbyte/actions/runs/11618002737/job/32354454829
  - Successful with no record mismatches
  - There was an uptick in the number of state messages going from 7 ->29. This is also expected because the concurrent framework has built in guarantees that at least one state message is emitted. We had a small regression in the legacy CDK where we accidentally dedupe state messages and the previous chargebee source which is now fixed in concurrent

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/manifest.yaml [hunk 1]
```diff
@@ -25,6 +25,7 @@ definitions:
               action: IGNORE
               error_message: "Stream is available only for Product Catalog 1.0"
         - type: DefaultErrorHandler
+          max_retries: 10
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-chargebee/source_chargebee/manifest.yaml:28]: My understanding was the default was 5. Why increase this to 10?

File: airbyte-integrations/connectors/source-chargebee/unit_tests/resource/config/config.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-chargebee/unit_tests/resource/config/config.json:1]: Why was this added?

**Review by maxi297 — APPROVED** [review]
Approving as my comments are just for me to understand and I expect them not to cause any issue

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/manifest.yaml (hunk 1 above)
Comment (brianjlai) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/manifest.yaml:28]: the idea here was that due to us having a higher level of concurrency, we have a higher chance of getting rate limited. This would make the sync more resilient and when we incrementally back off we should get to a point where we can retry since the limit resets per minute.

On a tangent, I also have been thinking whether having a `max_retries` should even exist at all. Aside from the cost of extra resourcing, it feels like a sync should never crash unless the container crashed or the heartbeat times out.

File: airbyte-integrations/connectors/source-chargebee/unit_tests/resource/config/config.json (hunk 2 above)
Comment (brianjlai) [reply airbyte-integrations/connectors/source-chargebee/unit_tests/resource/config/config.json:1]: Good question. We didn't hit the required threshold for code coverage so I added some tests to verify that `run.py` could run a check and I needed a `config.json` fixture to do so

File: airbyte-integrations/connectors/source-chargebee/source_chargebee/manifest.yaml (hunk 1 above)
Comment (maxi297) [reply airbyte-integrations/connectors/source-chargebee/source_chargebee/manifest.yaml:28]: I partially agree. I think what concerns me is infinite loops and the fact that we might not be notified if there is an issue as we can't have an alert until the source ends with a code different than 0.
